src/components/cax-companyData.tsx

```diff
@@ -222,7 +222,7 @@ export const CompanyDataCax = () => {
   const validateLegalEntity = (value: string) => {
     setLegalEntity(value)
 
-    if (!Patterns.legalEntityPattern.test(value)) {
+    if (!Patterns.legalEntityPattern.test(value?.trim())) {
```

Comment:
personally I'd prefer to replace all the `Patterns.xyz.test(value)` with `isXYZ` - here `isLegalEntity` just like `isBPN` above. Makes the code more readable.